### ISSUE TITLE
SDK: Allow custom commands to be executed when server starts/stops

### ISSUE THREAD

**Issue by nihonium-cfx** [citizenfx/fivem#1103]
Indeed, would be great to have this when dealing with external database services used by some resources.

It means built project won't be self-contained anymore, though.

See: https://forum.cfx.re/t/method-of-launching-an-external-application-together-with-the-server/4773797

**Comment by blattersturm** [comment]
I don't think this is a good idea- rather, the project should be able to bundle such dependencies (such as downloaded DBMS binaries for a target platform) and run them in an isolated scope (e.g. named pipe, or when not supported a local-scope TCP port) for cases like this.

This should also allow handling e.g. public resources using MySQL without requiring the user to bother deploying their 'own' database install.

As to @Jonathan-Rosewood's comment, I don't think supporting custom deployment flow is a good idea, at least not until we have a deployment framework built in. Nothing is as permanent as a temporary solution!

**Comment by Jonathan-Rosewood** [comment]
Fair enough, but then the whole idea sounds like the invention of the bicycle, if I understood everything correctly (my English is not so good, maybe)

For example, it's easier to use Docker to run and deploy a project with all dependencies, proper isolation, and close to the real environment, at least for development purposes.

But the current implementation of FXDK does not allow us to go beyond the scope of its build / launch functionality.

**Comment by blattersturm** [comment]
> For example, it's easier to use Docker to run and deploy a project with all dependencies, proper isolation, and close to the real environment, at least for development purposes.

If you're using 'Docker', you're way out of scope for this anyway. In addition, 'Docker' has no meaningful equivalent on Windows other than some shady commercial offering.
